src/components/header/SearchInput.tsx

```diff
@@ -0,0 +1,57 @@
+import { ReactComponent as Search } from '@assets/svg/search.svg';
+import {
+  Divider,
+  IconButton,
+  InputBase,
+  MenuItem,
+  Paper,
+  Select,
+} from '@mui/material';
+import { useGetCategoriesQuery } from '@services/ecommerce';
+import {
+  searchInput,
+  searchInputButton,
+  searchInputDivider,
+  searchInputField,
+  searchInputSelect,
+} from '@styles/header-styles';
+import { useState } from 'react';
+import { SelectArrowIcon } from './SelectArrowIcon';
+
+export default function SearchInput() {
+  const [selectState, setSelectState] = useState('All Categories');
+  const { data: categories } = useGetCategoriesQuery('');
```

Comment:
<img width="534" alt="Screenshot 2022-10-07 at 15 32 55" src="https://user-images.githubusercontent.com/64188262/194554226-75afe9d0-7f02-489c-824e-899967eb571e.png">
You don't pass any arguments, so you do not need to pass an empty string to your useGetCategoriesQuery()